Source/WebKit/UIProcess/API/gtk/WebKitPrintOperation.cpp

```diff
@@ -393,6 +424,222 @@ static void webkitPrintOperationPrintPagesForFrame(WebKitPrintOperation* printOp
 #endif
 }
 
+static GRefPtr<GtkPrinter> findFilePrinter(GtkPrintSettings* settings)
+{
+    struct FindPrinterData {
+        GRefPtr<GtkPrinter> printer;
+    } data = { nullptr };
+
+    gtk_enumerate_printers([](GtkPrinter* printer, gpointer userData) -> gboolean {
+        auto& data = *static_cast<FindPrinterData*>(userData);
+        auto* backend = gtk_printer_get_backend(printer);
+        if (!g_strcmp0(G_OBJECT_TYPE_NAME(backend), "GtkPrintBackendFile")) {
+            data.printer = printer;
+            return TRUE;
+        }
+        return FALSE;
+    }, &data, nullptr, TRUE);
+
+    return data.printer;
+}
+
+static void webkitPrintOperationSendPagesToPrintPortal(WebKitPrintOperation* printOperation, WebFrameProxy* webFrame, GtkPrintSettings* printSettings, GtkPageSetup* pageSetup)
+{
+    auto* priv = printOperation->priv;
+
+    RELEASE_ASSERT(priv->portalProxy);
+    RELEASE_ASSERT(priv->preparePrintResponse && priv->preparePrintResponse->token());
+    RELEASE_ASSERT(priv->preparePrintResponse->result() == WEBKIT_PRINT_OPERATION_RESPONSE_PRINT);
+
+    const char* applicationName = g_get_application_name();
+    GUniquePtr<char> jobName(g_strdup_printf("%s job #%u", applicationName ? applicationName : "WebKit", ++jobNumber));
+    GRefPtr<GtkPrinter> filePrinter = findFilePrinter(printSettings);
+
+    // Print the page into a temporary file
+    GUniqueOutPtr<char> filename;
+    int fd = g_file_open_tmp("webkitgtkprintXXXXXX", &filename.outPtr(), nullptr);
+    close(fd);
+    GUniquePtr<char> uri(g_filename_to_uri(filename.get(), NULL, NULL));
```

Comment:
```suggestion
    GUniquePtr<char> uri(g_filename_to_uri(filename.get(), nullptr, nullptr));
```

Another difference you'll get used to.

That said, please check for errors here too. It's hard to predict what all might go wrong. E.g. maybe the filename contains directory separator characters. Who knows.